hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -331,6 +333,19 @@ protected final void saveInternalSchema(HoodieTable table, String instantTime, H
 
   protected abstract HoodieTable<T, I, K, O> createTable(HoodieWriteConfig config, Configuration hadoopConf, HoodieTableMetaClient metaClient);
 
+  /**
+   * Validate timestamp for new commits.
+   * @param metaClient instance of{@link HoodieTableMetaClient} to be used.
+   * @param instantTime instant time of the current commit thats in progress.
+   */
```

Comment:
Can we address the semancis in the doc, for e.g, what scenarios are good to go and what are unexpected.